### PR TITLE
audacious: enable Qt interface

### PR DESCRIPTION
**Pull request by bra1nwave**
closes  https://github.com/void-linux/void-packages/issues/14171

Should I add a build option for Qt/GTK?

### CODE REVIEW

**Comment by Piraty** [comment]
I think a built option would be good

**Comment by bra1nwave** [comment]
Done

**Comment by Chocimier** [comment]
Default build depends on both qt and gtk now. Can they be split out to subpkgs?

**Comment by xtraeme** [comment]
+1

**Comment by bra1nwave** [comment]
> Default build depends on both qt and gtk now. Can they be split out to subpkgs?

Yeah, I think so. Then the build options are redundant, right?

**Comment by Chocimier** [comment]
Yes, no need for build options.

**Comment by bra1nwave** [comment]
It's not easy as I thought it would be. The main culprit here is `audacious-plugins`. It has a lot of various .so plugins which either depend on GTK or QT. Sure, we can hunt down every SO, but this would be a nightmare to maintain. 

There is also the problem that some plugins that were not previously dependent on Qt are now depending on it. Which means I cannot move it into a subpkg.

We have (IMHO) two alternatives:
- Keep the Build options and make GTK the default one. If someone only needs the Qt version, he/she can build it with `-o ~gtk,qt`
- Make separate (`audacious-qt` and `audacious-qt-plugins`) templates with disabled GTK

**Comment by Piraty** [comment]
You could also do two builds in the template (one for each variant toolkit) and split results in corespoinding subpackages, which is not a nice thing either. Also dependency-overwriting if the shared package is possible, but even worse.

I'd say: only support one toolkit and go for a default build option. If demand should arise to provide both versions at the same time, we can still warmup this discussion.

**Comment by bra1nwave** [comment]
@Piraty Done

File: srcpkgs/audacious/template [hunk 1]
```diff
@@ -14,6 +15,15 @@ homepage="https://audacious-media-player.org/"
 distfiles="https://distfiles.${pkgname}-media-player.org/${pkgname}-${version}.tar.bz2"
 checksum=8366e840bb3c9448c2cf0cf9a0800155b0bd7cc212a28ba44990c3d2289c6b93
 
+build_options="gtk qt"
+build_options_default="gtk"
+desc_option_gtk="Enable support for the GTK+2 GUI toolkit"
+vopt_conflict gtk qt
+
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" qt5-host-tools"
```
Comment (Chocimier) [inline srcpkgs/audacious/template]: Only needed with qt build option, can be added regardless of cross build.

**Pull request merged**